apps/prairielearn/elements/pl-number-input/pl-number-input.py

```diff
@@ -294,7 +311,7 @@ def render(element_html, data):
                 else:
                     html_params["incorrect"] = True
             except Exception:
-                raise ValueError("invalid score" + score)
```

Comment:
For this, the following badge-setting logic can be replaced with the helper function `determine_score_params` (https://github.com/PrairieLearn/PrairieLearn/blob/master/apps/prairielearn/python/prairielearn.py#L138). Take a look at other elements that use this helper function for examples.

Reply:
Fixed. I'm surprised only 3 other elements currently use this.

Reply:
@Frozemint You don't need to put this in a try-except block anymore, and you should also use this function on line 141 (where the score is set as well).

Reply:
Thanks for catching that. Is there anything else that is missing in this PR?

Reply:
Outside of changes waiting for the integer input PR, I think this is ok for now. There's a change I might want to make to the feedback that gets displayed to students, but I need to think about it first.

Reply:
@Frozemint looking at this again (and since your other PR got closed out), I think it would be good to address https://github.com/PrairieLearn/PrairieLearn/issues/710 as part of this PR, and change the way that correct answer feedback is displayed / processed to match that of the units input element (for example). That way, the additional feedback in the linked issue can be a short addon to the grader code. Feel free to DM me if you would like a more detailed description.

Also, great work on your PRs so far!

Reply:
*shot

Reply:
@eliotwrobson I took a look at `pl-unit-input` and the way they handle feedback (correct unit, incorrect magnitude), and added something similar in the commits I just pushed. Since neither `pl-number-input` or `pl-unit-input` uses JS, this implementation is not JS based. My implementation is quite crude and quickly put together in case this isn't quite what you are looking for. Please have a look and feel free to make any changes you see fit.

Reply:
@Frozemint thanks! I think this is a really good starting point. Sorry the issue was a bit unclear, we hashed out some discussion in the comments section about how the precision should be handled. This was exactly how you did it though (on the backend in Python and not in JS). There are a couple of minor notes I have, will leave in a review.